### ISSUE TITLE
Question about the MIL model Training

### ISSUE THREAD

**Issue by sketchplanet** [ThoroughImages/CAMEL#4]
Hi
Thanks for providing this great paper and dataset. 
1) I found that the training of the MIL model need spend a lot of time because before the instance selection in every step you must inference the whole tiles in your training set how do you deal with it ?
2) At the beginning of the training of MIL, the weight of the model are random so the selection of the tile is also random. Will you pretrain the model in the noisy dataset(Labeling each tile with the Slide's label) then train the MIL model to make the selection of the tile at beginning of MIL training procedure meaningful ?

**Comment by physicso** [comment]
Thank you very much for your interest in our work.

1. The training stage of the MIL model does not need extra inference. During training, similar to regular classifiers, each instance from an image undergoes an inference phase. Different from regular classifiers, where all instances contribute to the loss function calculated by e.g., the cross-entropy, only one instance will be considered in the loss in MIL. This may lead to gradient drift in the training process. To overcome, we need to train the cMIL model with more images, thus multiple GPUs are required.

2. Although we have not tested this approach, we guess using the pre-trained model from the noisy dataset may not be very helpful since many false-positive instances emerge, which may lead the training procedure to a wired situation.

**Comment by physicso** [comment]
"To overcome, we need to train the cMIL model with more images." We mean we need to feed more images in each step, while, for instance, each image occupies one GPU :)

**Comment by sketchplanet** [comment]
I got it. Thank you for the reply ! 
A question still bothers me. If the weight of cMIL model initializes randomly(or pretrained with imagenet) and the tile will be selected randomly. If the tiles are selected in Non Cancerous region in the Cancerous slides(the probs depends on the ratio of  CA region on the slides), the model is more inclined to select NC tiles on  CA slides in the next training step  which make the model worsen.  I'm curious whether the convergence of the training of cMIL model is influenced by it in your experiment?    I come cross this phenomenon when I apply the max-min MIL in other data sets(The resolution of the image is 5000*5000 which has more tiles than yours mentioned in the paper ).

**Comment by physicso** [comment]
As demonstrated in our paper, MAX-MIN criterion leads to the overfitting (false-positive) problem. This is why we combined MAX-MIN and MAX-MAX together in the cMIL training stage. 

Meanwhile, as I said before, multiple GPUs are recommended to relieve the gradient drift problem.

Also, if the image size is large, we recommend you to try cascade data enhancement approach proposed in our paper.

**Comment by sketchplanet** [comment]
Thank you!
